File: 728-self-dividing-numbers/self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        res = []

        for number in range(left, right+1):
            isSelfDividing = True
            numString = str(number)
            digits = [int(char) for char in numString]

            for digit in digits:
                if digit == 0 or number % digit != 0:
                    isSelfDividing = False
                    break

            if isSelfDividing:
                res.append(number)


        return res

File: 728-self-dividing-numbers/self-dividing-numbers.swift
class Solution {
    func selfDividingNumbers(_ left: Int, _ right: Int) -> [Int] {

        var arr: [Int] = []

        for number in left...right{
            let numStr = String(number)
            let numArr = numStr.compactMap { Int(String($0)) }
            var isSelfDividing = true

            for digit in numArr{
                if digit == 0 || number % digit != 0 {
                    isSelfDividing = false
                    break
                }
                
            }
            if isSelfDividing {
                arr.append(number)
            }
        }

        return arr
        
    }
}